[POS Settings] Hardware section navigation

iamgabrielma opened the pull request:
Closes WOOMOB-1039

## Description

This PR adds the basic scaffolding and navigation for the Hardware section of POS settings, by allowing the merchant to navigate through card reader and barcode scanner setup in a detail view to the right, as well as see documentation full screen (for now) presented on top of the view.

Final UI is not handled yet, this will be done separately.

https://github.com/user-attachments/assets/78343ca6-14d6-4bbb-86d8-a6784bf90773

## Testing information
- In POS, tap on the `...` CTA, and go to `Settings`
- Tap `Hardware` > `Card Readers`. Observe that you can open documentation full screen, and the last known reader model will appear if any.
- Tap `Hardware` > `Barcode scanner` . Observe you can trigger the barcode scanner setup, as well as see the documentation.

iamgabrielma committed:
set sidebar to 35% size

iamgabrielma committed:
hardware sub-navigation

iamgabrielma committed:
present modal and webview for hardware

iamgabrielma committed:
make cardReadersView, remove chevrons

iamgabrielma committed:
fetch and display last known reader

iamgabrielma committed:
present documentation full screen

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHardwareDetailView.swift:
```diff
@@ -20,19 +25,242 @@ struct PointOfSaleSettingsHardwareDetailView: View {
+            SafariView(url: WooConstants.URLs.inPersonPaymentsLearnMoreWCPay.asURL())
         }
+        .task { @MainActor in
+            // TODO: WOOMOB-1172
```
This is temporary, most likely we'll fetch card reader information from passing in either a reference to the aggregate model or expose its card reader details.

staskus replied:
👍 Yes, the aggregate model works.

If we think that this state sharing in an aggregate model is not needed, and we only need information within this view,  we could rely on a separate view model for this view that fetches information. 

I had the same question for orders, and at least for order it felt like it made sense to have something separate and there was no value putting information within the aggregate model.

iamgabrielma replied:
Jinx! I just had the same question when reviewing your PR :D 

I also think that in this case, for settings, it makes more sense to go through the aggregate, as we'll deal with settings across different features that are part of such aggregate, but we'll see what fits better 👍

staskus replied:
Yes, I agree. 👍

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsView.swift:
```diff
@@ -14,73 +14,79 @@ struct PointOfSaleSettingsView: View {
 
 extension PointOfSaleSettingsView {
+    enum Constants {
+        static let sidebarWidthFraction: CGFloat = 0.35
```
I think this is one of those we can share between Dashboard/Orders/Settings when we decide if we need a unified look and functionality.

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsView.swift:
```diff
@@ -14,73 +14,79 @@ struct PointOfSaleSettingsView: View {
-                default:
-                    EmptyView()
+                .frame(width: geometry.size.width * Constants.sidebarWidthFraction)
+                Group {
```
Wwe could avoid using `Group` if you would move the switch into a separate `@ViewBuilder`:

```swift
    @ViewBuilder
    private var detailView: some View {
        switch selection {
        case .store:
            PointOfSaleSettingsStoreDetailView()
        case .hardware:
            PointOfSaleSettingsHardwareDetailView()
        case .help:
            PointOfSaleSettingsHelpDetailView()
        default:
            EmptyView()
        }
    }
```

iamgabrielma replied:
Done! https://github.com/woocommerce/woocommerce-ios/pull/16038/commits/2098428e15a7d086dde3678260ce5ba713c1dc62

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHardwareDetailView.swift:
```diff
@@ -20,19 +25,242 @@ struct PointOfSaleSettingsHardwareDetailView: View {
+                    cardReadersView
+                case .hardware(.scanners):
+                    scannersView
+                case .scanner:
```
Could we remove `scanner` from the `NavigationDestination` then?

iamgabrielma replied:
Yes, I thought would be needed but it isn't since only buttons are there now. Updated: https://github.com/woocommerce/woocommerce-ios/pull/16038/commits/9fdcf3635a420db28939b49820634c86e1012e45

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHardwareDetailView.swift:
```diff
@@ -20,19 +25,242 @@ struct PointOfSaleSettingsHardwareDetailView: View {
+}
+
+extension PointOfSaleSettingsHardwareDetailView {
+    enum NavigationDestination: Hashable {
+        case hardware(PointOfSaleSettingsView.HardwareDestination)
+        case scanner(ScannerDestination)
+    }
```
Could be removed

```suggestion
    enum NavigationDestination: Hashable {
        case hardware(PointOfSaleSettingsView.HardwareDestination)
    }
```

iamgabrielma replied:
https://github.com/woocommerce/woocommerce-ios/pull/16038/commits/9fdcf3635a420db28939b49820634c86e1012e45

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHardwareDetailView.swift:
```diff
@@ -1,12 +1,17 @@
 
 struct PointOfSaleSettingsHardwareDetailView: View {
-    @State private var navigationPath: [PointOfSaleSettingsView.HardwareDestination] = []
+    @State private var navigationPath: [NavigationDestination] = []
```
In a broader conversation, I think the fact that "< Back" button is displayed within the details view, the "X" (Close) should be presented within the the list view.

And we likely need to use `POSPageHeaderView` and not the native navigation bar, and of course, not both. As I understand, we leave this for later 👍 

<img width="2420" height="1668" alt="Simulator Screenshot - iPad Pro 11-inch (M4) - 2025-08-26 at 14 11 48" src="https://github.com/user-attachments/assets/efec57ee-c6b3-4c26-9955-725c3bccbb8f" />

iamgabrielma replied:
> In a broader conversation, I think the fact that "< Back" button is displayed within the details view, the "X" (Close) should be presented within the the list view. And we likely need to use POSPageHeaderView and not the native navigation bar, and of course, not both. As I understand, we leave this for later 👍

Yes, that's something I'm precisely discussing today, as Android currently does not show a header across the full view (separates list and detail view), nor `X` to close the view that I can see. Neither this has been provided in the initial design drafts p1756257438170999?thread_ts=1755784099.956039&cid=C070SJRA8DP-slack-C070SJRA8DP

I think could even make sense to get the "Help" card up with the rest, and take that bottom-left position to go back to the dashboard, let's see what design thinks 👍

staskus reviewed (approved):
👍

iamgabrielma committed:
extract detailView

iamgabrielma committed:
remove scanner as nav destination

At the moment the scanner child view only has buttons that will prompt modal flow or web view, so no need to be part of the nav stack further down.

iamgabrielma committed:
move HardwareDestination to concrete view
